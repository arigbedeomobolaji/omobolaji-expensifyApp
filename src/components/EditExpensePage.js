//jshint ignore:start
import React from "react"
import { connect } from "react-redux"
import ExpenseForm from "./ExpenseForm"
import {editExpense, asyncRemoveExpense} from "../actions/expenses.action"

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    // props.dispatch(editExpense(props.expense.id, expense))
    this.props.editExpense(this.props.expense.id, expense)
    this.props.history.push("/")
  }

  handleRemoveExpense = (e) => {
    this.props.asyncRemoveExpense(this.props.expense.id)
    this.props.history.push("/")
  }

  render() {
    return (
      <div>
        <h1>Expensify App | Edit</h1>
      <ExpenseForm
        expense={this.props.expense}
        onSubmit={this.onSubmit}
      />
      <button onClick={this.handleRemoveExpense}
      >Remove</button>
      </div>
    )
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => expense.id === props.match.params.id)   
})

const mapDispatchToProps = (dispatch) => ({
  editExpense: (id, expense) => {
    dispatch(editExpense(id, expense))
  },
  asyncRemoveExpense: (id) => {
    dispatch(asyncRemoveExpense(id))
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)
