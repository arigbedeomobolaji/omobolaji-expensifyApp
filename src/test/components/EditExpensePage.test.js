// jshint ignore:start
import React from "react"
import { shallow } from "enzyme"
import { EditExpensePage } from "../../components/EditExpensePage.js"
import expenses from "../fixtures/expense"

let onSubmit, history, asyncRemoveExpense, editExpense, wrapper
beforeEach(() => {
 // onSubmit = jest.fn()
 editExpense = jest.fn()
 history = { push: jest.fn() }
 asyncRemoveExpense = jest.fn()
 wrapper = shallow(<EditExpensePage editExpense={editExpense} asyncRemoveExpense={asyncRemoveExpense} history={history} expense={expenses[1]}/>)
})

test("should render expense page correctly", () => {
 expect(wrapper).toMatchSnapshot()
})

test("should handle remove expense", () => {
 wrapper.find("button").simulate("click")
 expect(history.push).toHaveBeenLastCalledWith("/")
 expect(asyncRemoveExpense).toHaveBeenLastCalledWith(expenses[1].id)
})

test("Should handle editExpense", () => {
 wrapper.find("ExpenseForm").prop("onSubmit")(expenses[1])
 expect(history.push).toHaveBeenLastCalledWith("/")
 expect(editExpense).toHaveBeenLastCalledWith(expenses[1].id, expenses[1])
})