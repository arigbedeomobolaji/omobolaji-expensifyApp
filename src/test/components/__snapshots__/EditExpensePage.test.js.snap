// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense page correctly 1`] = `
<div>
  <h1>
    Expensify App | Edit
  </h1>
  <ExpenseForm
    expense={
      Object {
        "amount": 100,
        "createdAt": -345600000,
        "description": "Gum",
        "id": "2",
        "note": "My chewing Gum for reading",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
